yapapi/network.py

```diff
@@ -261,6 +261,21 @@ async def add_node(self, node_id: str, ip: Optional[str] = None) -> Node:
 
         return node
 
+    async def refresh_nodes(self):
+        for node in self._nodes.values():
```

Comment:
Is there a reason to loop over nodes one by one?
Maybe instead:
```
async def refresh_nodes(self):
    async def refresh_node(node):
        try:
            await self._net_api...
         (...)
     await asincio.gather(refresh_node(n) for n in self._nodes.values())
```
I guess this would be faster and should be quite harmless?

Reply:
hmm... you may be right